### PR TITLE
[Buttons] [Ripple] fix button to update ripple states correctly

### PR DESCRIPTION
**Pull request by galiak11**
### Description
Buttons do not change the ripple state correctly when their selected or highlighted states are updated programmatically. To fix this issue, we're updating the RippleViews's highlighted and selected states to match the button's highlighted and selected states.

Steps to reproduce the highlighted state issued:
1. set: button.isHighlighted = true 
Expected: button is highlighted (background is light blue)
Outcome: button is not highlighted (background is white)

Steps to reproduce the selected state issued:
1. set: button.isSelected = true 
Expected: button is selected (background is light blue)
Outcome: button is not selected (background is white)

Before: 
image

After:
image

### Issue
b/128908676
Issue: https://github.com/material-components/material-components-ios/issues/7342

### CODE REVIEW

File: components/Buttons/src/MDCButton.m [hunk 1]
```diff
@@ -379,13 +379,13 @@ - (void)setEnabled:(BOOL)enabled animated:(BOOL)animated {
 
 - (void)setHighlighted:(BOOL)highlighted {
   [super setHighlighted:highlighted];
-
+  self.rippleView.rippleHighlighted = highlighted;
```
Comment (codeman7) [inline components/Buttons/src/MDCButton.m:377]: _Tip:_ Please add a new line above.

File: components/Buttons/src/MDCButton.m [hunk 2]
```diff
@@ -379,13 +379,13 @@ - (void)setEnabled:(BOOL)enabled animated:(BOOL)animated {
...
   [self updateAfterStateChange:NO];
 }
 
 - (void)setSelected:(BOOL)selected {
   [super setSelected:selected];
-
+  self.rippleView.selected = selected;
```
Comment (codeman7) [inline components/Buttons/src/MDCButton.m:384]: _Tip:_ Please add a new line above.

**Review by codeman7 — DISMISSED** [review]

**Review by jverkoey — DISMISSED** [review]
Please:

- [x] Provide repro steps.
- [x] Add before/after gifs to the issue description
- [x] Link to an open issue on Github that describes the incorrect behavior.

**Review by yarneo — DISMISSED** [review]
Hi Galia,

Can you please remove all other instances of setting self.rippleView.rippleHighlighted to a value. It should only be set in the setHighlighted of the UIControl ideally, unless the component is doing something unexpected. I would remove the calls to rippleHighlighted in the touchesBegan touchesEnded and other places. After that, please test and verify it is still working correctly.

**Review by randallli — COMMENTED** [review]
Let see if we can try and get some unit tests that this PR's description gives so that we wont regress on this in the future.

**Review by yarneo — APPROVED** [review]

**Pull request merged**